src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -315,22 +298,47 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
   }
 
   @Override
-  public ResponseEntity<JobModel> createGoogleContext(
-      UUID id, @Valid CreateGoogleContextRequestBody body) {
+  public ResponseEntity<JobReport> createCloudContext(
+      UUID id, @Valid CreateCloudContextRequest body) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    ControllerValidationUtils.validateCloudContext(body.getCloudContext());
+    String jobId = body.getJobControl().getId();
+    String resultRelativePath = getAsyncResultEndpoint(jobId);
+
+    workspaceService.createGoogleContext(id, jobId, resultRelativePath, userReq);
+    JobReport jobReport = jobService.retrieveJob(jobId, userReq);
```

Comment:
Moved to read the status code from the JobReport instead, and also changed the return type since it could be a `CreateCloudContextResult` instead of just a `JobReport`.